src/PointwiseFunctions/AnalyticSolutions/WaveEquation/IsotropicSphericalWave.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines ScalarWave::Solutions::IsotropicSphericalWave
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace ScalarWave {
+struct Pi;
+struct Psi;
+template <size_t Dim>
+struct Phi;
+}  // namespace ScalarWave
+namespace Tags {
+template <typename Tag>
+struct dt;
+}  // namespace Tags
+template <size_t VolumeDim>
+class MathFunction;
+
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ScalarWave {
+namespace Solutions {
+/*!
+ * \brief An isotropic spherical wave solution to the Euclidean wave equation
+ *
+ * The solution is given by \f$\Psi(\vec{x},t) = \Psi(r,t) =
+ * \frac{1}{r}F(r+t)+\frac{1}{r}G(r-t)\f$ where \f$F\f$ and
+ * \f$G\f$ are arbitrary one-dimensional functions. \f$F\f$ and
+ * \f$G\f$ represent ingoing and outgoing solutions of the one-dimensional wave
+ * equation, respectively.
+ *
+ * \tparam Dim The spatial dimension of the solution. Currently only implemented
+ * for 3 spatial dimensions.
+ */
+template <size_t Dim>
+class IsotropicSphericalWave {
+ public:
+  struct IngoingProfile {
+    using type = std::unique_ptr<MathFunction<1>>;
+    static constexpr OptionString help = {"The profile of the ingoing wave."};
+  };
+  struct OutgoingProfile {
```

Comment:
The ingoing and outgoing profile must be the same otherwise the wave isn't regular.